ksql-engine/src/main/java/io/confluent/ksql/function/UdfTemplate.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import com.google.common.primitives.Primitives;
+import com.squareup.javapoet.CodeBlock;
+import java.lang.reflect.Method;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("WeakerAccess") // used from generated code
```

Comment:
nit: not sure this is needed. I think you can just make `generateCode` package local ...

Reply:
True for the `generateCode` method, but I need the `coerce` method to be truly public (and therefore the class). I updated `generateCode`